tests/integration/azure/test_access.py

```diff
@@ -37,6 +43,57 @@ def test_save_spn_permissions_local(ws, sql_backend, inventory_schema, make_rand
     sql_backend.save_table(f"{inventory_schema}.external_locations", tables, ExternalLocation)
     location = ExternalLocations(ws, sql_backend, inventory_schema)
     installation = Installation(ws, make_random(4))
-    az_res_perm = AzureResourcePermissions(installation, ws, AzureResources(ws, include_subscriptions=""), location)
+    azure_mgmt_client = AzureAPIClient(
+        ws.config.arm_environment.resource_manager_endpoint,
+        ws.config.arm_environment.service_management_endpoint,
+    )
+    graph_client = AzureAPIClient("https://graph.microsoft.com", "https://graph.microsoft.com")
+    azure_resources = AzureResources(azure_mgmt_client, graph_client)
+    az_res_perm = AzureResourcePermissions(installation, ws, azure_resources, location)
     path = az_res_perm.save_spn_permissions()
     assert ws.workspace.get_status(path)
+
+
+@pytest.mark.skip
+def test_create_global_spn(ws, sql_backend, inventory_schema, make_random, make_cluster_policy):
+    tables = [
+        ExternalLocation("abfss://things@labsazurethings.dfs.core.windows.net/folder1", 1),
+    ]
+    sql_backend.save_table(f"{inventory_schema}.external_locations", tables, ExternalLocation)
+    location = ExternalLocations(ws, sql_backend, inventory_schema)
+    installation = Installation(ws, make_random(4))
+    policy = make_cluster_policy()
+    installation.save(WorkspaceConfig(inventory_database='ucx', policy_id=policy.policy_id))
+    azure_mgmt_client = AzureAPIClient(
+        ws.config.arm_environment.resource_manager_endpoint,
+        ws.config.arm_environment.service_management_endpoint,
+    )
+    graph_client = AzureAPIClient("https://graph.microsoft.com", "https://graph.microsoft.com")
+    azure_resources = AzureResources(azure_mgmt_client, graph_client)
+    az_res_perm = AzureResourcePermissions(installation, ws, azure_resources, location)
+    az_res_perm.create_uber_principal()
+    config = installation.load(WorkspaceConfig)
+    assert config.global_spn_id is not None
+    policy_definition = json.loads(ws.cluster_policies.get(policy_id=policy.policy_id).definition)
+    resource_id = "/subscriptions/bf472657-510a-4fad-a050-87cfb4e1a2ce/resourceGroups/HSRG/providers/Microsoft.Storage/storageAccounts/labsazurethings"
```

Comment:
Don't leak internal subscription ids, put resource id in debug env (or your IDE run configuration) and skip the test if it's not there 
```suggestion
    resource_id = env_or_skip("TEST_STORAGE_RESOURCE")
```